From 67fb2f1c041159809ea1f2656f19116b2f3a5434 Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Sat, 10 Feb 2024 12:55:08 +0100
Subject: [PATCH] rust toolchain: fix float target-abi for riscv64

Patch-Filename: 0001-rust-toolchain-fix-float-target-abi-for-riscv64.patch
---
 tools/rust/build_rust.py | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/tools/rust/build_rust.py b/tools/rust/build_rust.py
index 9b1acc5a5faf9..4160672118645 100755
--- a/tools/rust/build_rust.py
+++ b/tools/rust/build_rust.py
@@ -779,6 +779,18 @@ def main():
 
     if not args.skip_checkout:
         CheckoutGitRepo('Rust', RUST_GIT_URL, checkout_revision, RUST_SRC_DIR)
+        RunCommand([
+            'git', '-C', RUST_SRC_DIR, 'remote', 'add', 'kxxt',
+            'https://github.com/kxxt/rust.git'
+        ],
+                   fail_hard=False)
+        RunCommand([
+            'git', '-C', RUST_SRC_DIR, 'fetch', 'kxxt',
+                      'a2dfdaedc20d5fed8873e6e2c310c93431aa9c5e'
+        ])
+        RunCommand([
+            'git', '-C', RUST_SRC_DIR, 'cherry-pick',
+                      'a2dfdaedc20d5fed8873e6e2c310c93431aa9c5e'])
 
     if not args.update_deps:
         VerifyStage0JsonHash()
-- 
2.39.2

